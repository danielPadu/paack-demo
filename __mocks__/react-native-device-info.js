export default {
  getModel: jest.fn(),
  getVersion: jest.fn(),
  getBuildNumber: jest.fn(),
  getBundleId: jest.fn(),
  hasNotch: jest.fn(),
};
