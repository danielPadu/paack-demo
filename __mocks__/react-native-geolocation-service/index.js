export default {
  addListener: jest.fn(),
  getCurrentPosition: () => ({
    coords: {
      accuracy: 0,
      altitude: 0,
      altitudeAccuracy: 0,
      heading: 0,
      latitude: 0,
      longitude: 0,
      speed: 0,
    },
  }),
  removeListeners: jest.fn(),
  requestAuthorization: jest.fn(),
  setConfiguration: jest.fn(),
  startObserving: jest.fn(),
  stopObserving: jest.fn(),
};
