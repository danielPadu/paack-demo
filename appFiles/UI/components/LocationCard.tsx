import React, {useCallback, useState} from 'react';
import {
  ActivityIndicator,
  StyleProp,
  StyleSheet,
  Text,
  TextStyle,
  View,
} from 'react-native';
import {Button, Card, Colors} from 'react-native-ui-lib';
import {useSelector} from 'react-redux';
import useLocation from '../../hooks/useLocation';
import {RenderField} from './RenderField';

Colors.loadColors({
  primaryColor: '#2364AA',
  secondaryColor: '#81C3D7',
  textColor: '#221D23',
  errorColor: '#E63B2E',
  successColor: '#ADC76F',
  warnColor: '#FF963C',
});

export const LocationCard = () => {
  const {app_allow_geolocation, last_location} = useSelector(
    (state: any) => state.appReducer,
  );
  const {getLocation} = useLocation();
  const {
    cardContainer,
    cardTitleContainer,
    clientText,
    cardStyle,
    emptyContentText,
    refreshButtonContainer,
    changeStatusButtonStyle,
  } = styles;

  const [refreshingLoading, setRefreshingLoading] = useState(false);

  const getCoordsColor = () => ({
    color: 'green',
    fontSize: 12,
    fontWeight: '900',
  });

  const makeActiveLoadingIndicator = useCallback(() => {
    const {refreshButtonLoadingStyle} = styles;
    return refreshingLoading ? (
      <ActivityIndicator
        testID="activity-indicator"
        style={refreshButtonLoadingStyle}
        size="small"
        color="#0000ff"
      />
    ) : null;
  }, [refreshingLoading]);

  const onRefreshLocationPress = () => {
    setRefreshingLoading(true);
    getLocation().then(() => {
      setRefreshingLoading(false);
    });
  };

  return (
    <View style={cardContainer}>
      {app_allow_geolocation && last_location ? (
        <Card
          row={false}
          style={cardStyle}
          enableShadow={true}
          useNative
          activeOpacity={1}>
          <View style={cardTitleContainer}>
            <Text style={clientText}>
              {'Last location details registered '}
            </Text>
          </View>

          <RenderField title={'Accuracy: '} value={last_location?.accuracy} />
          <RenderField
            title={'Altitude: '}
            value={`${last_location?.altitude}`}
          />
          <RenderField
            title={'Altitude accuracy: '}
            value={`${last_location?.altitudeAccuracy}`}
          />
          <RenderField
            title={'Heading: '}
            value={`${last_location?.heading}`}
          />
          <RenderField title={'Speed: '} value={`${last_location?.speed} `} />

          <RenderField
            valueTestId={'latitude-value'}
            title={'Latitude: '}
            value={`${last_location?.latitude} `}
            valueExtraStyle={getCoordsColor() as StyleProp<TextStyle>}
          />
          <RenderField
            valueTestId={'longitude-value'}
            title={'Longitude: '}
            value={`${last_location?.longitude} `}
            valueExtraStyle={getCoordsColor() as StyleProp<TextStyle>}
          />

          <View style={refreshButtonContainer}>
            <Button
              testID={'refreshLocationButton'}
              label="Refresh location"
              disabled={refreshingLoading === true}
              onPress={onRefreshLocationPress}
              style={changeStatusButtonStyle}
              iconOnRight
              iconSource={makeActiveLoadingIndicator}
              bg-secondaryColor
              square
            />
          </View>
        </Card>
      ) : (
        <Text style={emptyContentText}>
          {'No location details were registered'}
        </Text>
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  cardContainer: {
    margin: 0,
    marginTop: 2,
    padding: 0,
    marginHorizontal: 15,
    justifyContent: 'center',
    alignItems: 'center',
    paddingVertical: 5,
  },
  cardTitleContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    width: '100%',
  },
  clientText: {
    alignItems: 'center',
    justifyContent: 'center',
    alignSelf: 'center',
    paddingVertical: 10,
    flex: 1,
    textAlign: 'center',
  },
  refreshButtonLoadingStyle: {marginLeft: 10},

  cardStyle: {
    flex: 1,
    marginTop: 5,
    padding: 10,
    flexDirection: 'column',
    justifyContent: 'flex-start',
    alignItems: 'center',
    width: '100%',
    minHeight: 300,
    overflow: 'hidden',
    borderRadius: 20,
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 6,
    },
    shadowOpacity: 0.39,
    shadowRadius: 8.3,

    elevation: 13,
  },

  emptyContentText: {
    textAlign: 'center',
    fontSize: 20,
    fontWeight: '900',
    color: 'black',
  },
  refreshButtonContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    width: '100%',
    marginVertical: 5,
  },
  changeStatusButtonStyle: {
    height: 50,
    paddingVertical: 0,
    justifyContent: 'center',
    alignItems: 'center',
    // marginLeft: 10,
  },
});
