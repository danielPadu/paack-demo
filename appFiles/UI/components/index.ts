export {RenderField} from './RenderField';
export {LocationCard} from './LocationCard';
