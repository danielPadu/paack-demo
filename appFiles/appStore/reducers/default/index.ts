import {appDefault} from './appReducerDefault';
import {modalDefault} from './modalReducerDefault';
export const defaultReducer = {
  appReducer: appDefault, //persistence: not to be in black list if settings should persist
  modalReducer: modalDefault,
};
