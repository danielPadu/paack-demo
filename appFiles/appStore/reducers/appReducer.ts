import {
  SET_ALLOW_GEOLOCATION,
  SET_ACTIVE_DELIVERY_ID,
  SET_ACTIVE_SCREEN,
  SET_LAST_LOCATION,
} from '../actions/actionTypes';
import {defaultReducer} from './default';

export default (
  state = defaultReducer.appReducer,
  action: {
    type: string;
    payload: boolean | string | typeof defaultReducer.appReducer.last_location;
  },
) => {
  switch (action.type) {
    case SET_ALLOW_GEOLOCATION: {
      return {...state, app_allow_geolocation: action.payload};
    }
    case SET_ACTIVE_DELIVERY_ID: {
      return {...state, active_delivery_id: action.payload};
    }
    case SET_ACTIVE_SCREEN: {
      return {...state, active_scren: action.payload};
    }
    case SET_LAST_LOCATION: {
      return {...state, last_location: action.payload};
    }
    default:
      return state;
  }
};
