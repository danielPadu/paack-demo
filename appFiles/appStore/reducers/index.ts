import {CombinedState, combineReducers} from 'redux';
import AppReducer from './appReducer';
import ModalReducer, {PayloadType} from './modalReducer';
import {RESET_STORE} from '../actions/actionTypes';
import {defaultReducer} from './default';
import FilesystemStorage from 'redux-persist-filesystem-storage';
export const reducerConfig = {
  timeout: 0, // The code base checks for falsy, so 0 disables
  key: 'root',
  storage: FilesystemStorage,
  whitelist: ['appReducer'],
  blacklist: ['modalReducer'],
};

const applicationReducer = combineReducers({
  appReducer: AppReducer,
  modalReducer: ModalReducer,
});

const RootReducer = (
  state:
    | CombinedState<{
        appReducer: typeof AppReducer;
        modalReducer: typeof ModalReducer;
      }>
    | undefined,
  action:
    | {
        type: string;
        payload:
          | string
          | boolean
          | typeof defaultReducer.appReducer.last_location;
      }
    | {type: string; payload?: PayloadType | undefined},
) => {
  let finalState = applicationReducer(state, action);
  if (action.type === RESET_STORE) {
    finalState = defaultReducer as typeof finalState;
  }
  return finalState;
};
export default RootReducer;
