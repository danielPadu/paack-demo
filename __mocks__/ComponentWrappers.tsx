import {NavigationContext} from '@react-navigation/native';
import React from 'react';
import {Provider} from 'react-redux/src';
import {PersistGate} from 'redux-persist/integration/react';
import {persistor} from '../appFiles/appStore/store';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import {defaultReducer} from '../appFiles/appStore/reducers/default';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
export const mockedStore = mockStore(defaultReducer);

export const navContext = {
  isFocused: () => true,
  // addListener returns an unscubscribe function.
  addListener: jest.fn(() => jest.fn()),
  navigate: Function.prototype,
  setParams: Function.prototype,
  dispatch: Function.prototype,
  getParam: (param: {deliveryId?: string} & any, defaultValue: any) => {
    return defaultValue;
  },
};

export const ComponentWrappers = (props: {
  children: React.ReactNode;
  redux_store: any;
}) => {
  const {children, redux_store} = props;

  return (
    <>
      <Provider store={redux_store ?? mockedStore}>
        <PersistGate loading={null} persistor={persistor}>
          <NavigationContext.Provider value={navContext as any}>
            {children}
          </NavigationContext.Provider>
        </PersistGate>
      </Provider>
    </>
  );
};
