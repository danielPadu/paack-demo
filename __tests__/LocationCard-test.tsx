import React from 'react';
import {fireEvent, render, waitFor} from 'react-native-testing-library';
import {useSelector} from 'react-redux';
import renderer, {
  act,
  ReactTestRenderer,
  ReactTestRendererJSON,
} from 'react-test-renderer';
import {LocationCard} from '../appFiles/UI/components/LocationCard';
import {ComponentWrappers, mockedStore} from '../__mocks__/ComponentWrappers';
import {defaultReducer} from '../appFiles/appStore/reducers/default';
let store;
let component:
  | ReactTestRenderer
  | ReactTestRenderer[]
  | ReactTestRendererJSON
  | ReactTestRendererJSON[]
  | null;
// test('renders correctly', async () => {
//   await act(async () => {
//     let store;
//     beforeEach(() => {
//       store = ms;
//       expect(
//         renderer
//           .create(
//             <ComponentWrappers redux_store={store}>
//               <LocationCard />
//             </ComponentWrappers>,
//           )
//           .toJSON(),
//       ).toMatchSnapshot();
//     });
//   });
// });

// it('renders correctly when there are no items', () => {
//   const tree = renderer
//     .create(
//       <ComponentWrappers redux_store={ms}>
//         <LocationCard />
//       </ComponentWrappers>,
//     )
//     .toJSON();
//   expect(tree).toMatchSnapshot();
// });
// test('checking props correctly', async () => {
//   await act(async () => {
//     let store;
//     let component;
//     beforeEach(() => {
//       store = mockStore([
//         {
//           appReducer: {
//             app_allow_geolocation: false,
//             active_delivery_id: '',
//             active_scren: '',
//             last_location: {
//               accuracy: 0,
//               altitude: 0,
//               altitudeAccuracy: 0,
//               heading: 0,
//               latitude: 0,
//               longitude: 0,
//               speed: 0,
//             },
//           },
//         },
//       ]);
//       component = render(
//         <ComponentWrappers redux_store={store}>
//           <LocationCard />
//         </ComponentWrappers>,
//       );
//       //expect(component).toMatchSnapshot();
//       expect(component).toHaveProperty('app_allow_geolocation');
//     });
//   });
// });
const mockDispatch = jest.fn();
jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useSelector: () => ({app_allow_geolocation: true, last_location: {}}),
  useDispatch: () => mockDispatch,
}));
describe('checking LocationCard component', () => {
  test('LocationCard internal functionallities', async () => {
    await act(async () => {
      await beforeEach(async () => {
        store = mockedStore;
        component = render(
          <ComponentWrappers redux_store={store}>
            <LocationCard />
          </ComponentWrappers>,
        );

        const {getByTestId, queryByTestId, toJSON} = component;
        const mockedHandler = jest.fn();

        const button = getByTestId('refreshLocationButton');
        button.props.onPress = mockedHandler;
        fireEvent.press(button);
        expect(button).toBeTruthy();
        // test('button pressed', () => {
        // await waitFor(() => expect(mockedHandler).toHaveBeenCalled());
        // });

        await waitFor(() =>
          expect(getByTestId('activity-indicator')).toBeTruthy(),
        );

        await waitFor(() =>
          expect(queryByTestId('longitude-value')).toBeTruthy(),
        );

        await waitFor(() =>
          expect(queryByTestId('longitude-value')).toBeTruthy(),
        );
        expect(toJSON()).toMatchSnapshot();
      });
      await afterEach(() => {
        // cleanup on exiting
        component?.remove?.();
        component = null;
      });
    });
  });
  test('LocationCard inner functionallity tests', async () => {
    await act(async () => {
      await beforeEach(async () => {
        store = mockedStore;
        component = renderer
          .create(
            <ComponentWrappers redux_store={store}>
              <LocationCard />
            </ComponentWrappers>,
          )
          .toJSON();
        const {findByProps} = component;
        const text = findByProps('Last location details registered ');

        await waitFor(() => expect(text).toBeTruthy());
        await waitFor(() => expect(component).toHaveProperty('cardContainer'));
      });
      await afterEach(() => {
        // cleanup on exiting
        useSelector?.mockClear?.();
        component?.remove?.();
        component = null;
      });
    });
  });
});
