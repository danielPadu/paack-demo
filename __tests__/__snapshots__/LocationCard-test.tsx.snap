// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checking LocationCard component LocationCard inner functionallity tests 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "margin": 0,
      "marginHorizontal": 15,
      "marginTop": 2,
      "padding": 0,
      "paddingVertical": 5,
    }
  }
>
  <View
    accessibilityState={
      Object {
        "selected": undefined,
      }
    }
    activeOpacity={1}
    delayPressIn={10}
    enableShadow={true}
    row={false}
    style={
      Array [
        undefined,
        undefined,
        undefined,
        undefined,
        Object {},
        Object {},
        Object {},
        Array [
          Object {
            "borderRadius": 12,
            "height": undefined,
            "overflow": "visible",
            "width": undefined,
          },
          Object {
            "borderRadius": 12,
          },
          Object {
            "elevation": 2,
          },
          Object {
            "shadowColor": "#A6ACB1",
            "shadowOffset": Object {
              "height": 5,
              "width": 0,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 12,
          },
          Object {
            "backgroundColor": "#FFFFFF",
          },
          undefined,
          Object {
            "alignItems": "center",
            "borderRadius": 20,
            "elevation": 13,
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "flex-start",
            "marginTop": 5,
            "minHeight": 300,
            "overflow": "hidden",
            "padding": 10,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 6,
              "width": 0,
            },
            "shadowOpacity": 0.39,
            "shadowRadius": 8.3,
            "width": "100%",
          },
        ],
      ]
    }
    useNative={true}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "flex": 1,
            "justifyContent": "center",
            "paddingVertical": 10,
            "textAlign": "center",
          }
        }
      >
        Last location details registered 
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "gray",
              "fontWeight": "600",
            }
          }
        >
           
          Accuracy: 
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 2,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "black",
                "fontWeight": "600",
                "textAlign": "center",
              },
              undefined,
            ]
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "gray",
              "fontWeight": "600",
            }
          }
        >
           
          Altitude: 
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 2,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "black",
                "fontWeight": "600",
                "textAlign": "center",
              },
              undefined,
            ]
          }
        >
          undefined
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "gray",
              "fontWeight": "600",
            }
          }
        >
           
          Altitude accuracy: 
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 2,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "black",
                "fontWeight": "600",
                "textAlign": "center",
              },
              undefined,
            ]
          }
        >
          undefined
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "gray",
              "fontWeight": "600",
            }
          }
        >
           
          Heading: 
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 2,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "black",
                "fontWeight": "600",
                "textAlign": "center",
              },
              undefined,
            ]
          }
        >
          undefined
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "gray",
              "fontWeight": "600",
            }
          }
        >
           
          Speed: 
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 2,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "black",
                "fontWeight": "600",
                "textAlign": "center",
              },
              undefined,
            ]
          }
        >
          undefined 
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "gray",
              "fontWeight": "600",
            }
          }
        >
           
          Latitude: 
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 2,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "black",
                "fontWeight": "600",
                "textAlign": "center",
              },
              Object {
                "color": "green",
                "fontSize": 12,
                "fontWeight": "900",
              },
            ]
          }
          testID="latitude-value"
        >
          undefined 
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "gray",
              "fontWeight": "600",
            }
          }
        >
           
          Longitude: 
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 2,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "black",
                "fontWeight": "600",
                "textAlign": "center",
              },
              Object {
                "color": "green",
                "fontSize": 12,
                "fontWeight": "900",
              },
            ]
          }
          testID="longitude-value"
        >
          undefined 
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginVertical": 5,
          "width": "100%",
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": true,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onLayout={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#D2D6D8",
            "borderRadius": 999,
            "flexDirection": "row",
            "height": 50,
            "justifyContent": "center",
            "minWidth": 90,
            "opacity": 1,
            "paddingHorizontal": 20,
            "paddingVertical": 0,
          }
        }
        testID="refreshLocationButton"
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "textAlign": "left",
              },
              undefined,
              undefined,
              undefined,
              undefined,
              Object {},
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "flex": 0,
                  "flexDirection": "row",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": "400",
                  "lineHeight": 24,
                },
                Object {
                  "color": "#FFFFFF",
                },
                Object {},
                Object {},
                undefined,
              ],
            ]
          }
          testID="refreshLocationButton.label"
        >
          Refresh location
        </Text>
        <ActivityIndicator
          color="#0000ff"
          size="small"
          style={
            Object {
              "marginLeft": 10,
            }
          }
          testID="activity-indicator"
        />
      </View>
    </View>
  </View>
</View>
`;
